types/validator_set.go

```diff
@@ -78,44 +78,45 @@ func (vals *ValidatorSet) IncrementProposerPriority(times int) {
 		panic("Cannot call IncrementProposerPriority with non-positive times")
 	}
 
-	const shiftEveryNthIter = 10
+	// cap the difference between priorities to be proportional to 2*totalPower by
+	// re-normalizing priorities, i.e., rescale all priorities by multiplying with:
+	//  2*totalVotingPower/(maxPriority - minPriority)
+	diff := computeMaxMinPriorityDiff(vals)
+	threshold := 2 * vals.TotalVotingPower()
+	if diff > threshold && threshold > 0 {
```

Comment:
Why would we ever have `threshold <= 0`

Reply:
As far as I remember, I've observed total voting power being 0 (at least in tests). Then, threshold would be 0, too.

Reply:
Ok. Shouldn't ever happen. Maybe there's some weird test where it does. I suppose it's fine for a sanity check but lets leave a note that it shouldn't happen and we ought to be able to remove that check

Reply:
E.g., the test `TestAveragingInIncrementProposerPriority` doesn't init voting powers. Only prios. as it only aims to test the priority averaging logic. But I remember there were other unrelated tests with voting power 0, too.